eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -37,24 +37,25 @@
                 "Windows2019": {
                     "OSVmImage": "MMS2019",
                     "Pool": "azsdk-pool-mms-win-2019-general",
-                    "BuildArgs": "--parallel 8",
                     "CMAKE_GENERATOR": "Visual Studio 16 2019"
                 }
             },
             "TargetPlatform": {
                 "Win32Api_curl": {
                     "VcpkgInstall": "curl[winssl] openssl",
-                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON"
+                    "CmakeArgs": " -DBUILD_TRANSPORT_CURL=ON",
+                    "BuildArgs": "--parallel 8"
                 },
                 "Win32Api_release_curl": {
                     "VcpkgInstall": "curl[winssl] openssl",
                     "CmakeArgs": " -DBUILD_TESTING=ON -DBUILD_PERFORMANCE_TESTS=ON -DRUN_LONG_UNIT_TESTS=ON -DBUILD_TRANSPORT_CURL=ON",
-                    "CMAKE_BUILD_TYPE": "Release",
+                    "BuildArgs": "--parallel 8 --config Release",
```

Comment:
Is there a reason you're limiting parallelism to 8 cores? Is there a separate option that says "use all the cores you have available" (similar to msbuild -m)?

Reply:
We are not limiting ( even though it looks like ).

I added this `--parallel 8` arg in the past because we were using the default, which was giving us 1 CPU and was taking longer.

The flag depends on the generator.  For Ninja, the default is to use all available CPUs, (but Ninja is no supported on CI).  For VStudio, default is 1, or whatever set with --parallel.
With make, default is 1, or whatever set in `-j `.  When no value is used for `-j`, make uses the MAX possible.
I am not sure if for Windows we can set `--parallel` only to ask for MAX

Reply:
For msbuild the -m switch says "use all the cores". -m:8 says "only use 8 cores".

Reply:
The build machines have only 2 cores. However, since the builds spend a lot of time waiting on I/O, it's usually faster to have more parallel units of build than cores up to a certain point. I believe someone did a performance test (I believe in DevOps) and determined that 8 was the point of diminishing marginal parallel returns.